common/src/main/java/com/wynntils/wc/utils/scoreboard/quests/QuestManager.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wc.utils.scoreboard.quests;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.mc.event.WynntilsScoreboardUpdateEvent;
+import com.wynntils.wc.utils.scoreboard.ScoreboardLine;
+import com.wynntils.wc.utils.scoreboard.ScoreboardManager;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import net.minecraft.ChatFormatting;
+import net.minecraft.server.ServerScoreboard;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.apache.commons.lang3.StringUtils;
+
+public class QuestManager {
+
+    private static final String TRACKED_QUEST_STRING = "§6§lTracked Quest:";
+
+    private static QuestInfo currentQuest = null;
+
+    @SubscribeEvent
+    public static void onScoreboardUpdate(WynntilsScoreboardUpdateEvent event) {
+        if (!event.getChangeMap().containsKey(WynntilsScoreboardUpdateEvent.ChangeType.Quest)) return;
+
+        Set<WynntilsScoreboardUpdateEvent.Change> changes =
+                event.getChangeMap().get(WynntilsScoreboardUpdateEvent.ChangeType.Quest);
+
+        for (WynntilsScoreboardUpdateEvent.Change change : changes) {
+            if (change.method() == ServerScoreboard.Method.REMOVE) {
+                if (TRACKED_QUEST_STRING.equals(change.line())) {
+                    questTrackingStopped();
+                }
+            } else { // Method is CHANGE
+                // Use copy constructor to prevent threading issues
+                List<ScoreboardLine> reconstructedScoreboard =
```

Comment:
I was about to add a comment on this in ScoreboardManager, but started by searching for use cases, and found this. :-) 

The problem is that you call `reconstructedScoreboard.sort()`, which will temporarily leave the scoreboard in a random state. If you sort it to a new List, and then just reassign the field, you will not have threading issues, and would not need to do a defensive copy here instead. (You'd still need to use `getReconstructedScoreboard()` and store the value in a local variable, so it does not change under your feet.)

Reply:
What if get method sorted the list and assigned it? Sort is not expensive if it is already sorted

Reply:
Oh actually your method is better, I get it now